### ISSUE TITLE
Enhancement: Throw a warning during `l3build ctan` if `l3build tag` would make changes

### ISSUE THREAD

**Issue by Skillmon** [latex3/l3build#222]
As I'm forgetting this frequently, and poor CTAN-maintainers always have to double check my packages, I think it would be nice (also for other users) to throw a warning during `l3build ctan` if `l3build tag` was forgotten (so if the usage of `l3build tag` would lead to different files).

**Comment by davidcarlisle** [comment]
I assume this would be by essentially doing l3build tag then git status or something to see what changed?
Currently l3build ctan (and l3build genera;;y) doesn't assume anything about about the version control in use.
There could be a buildctan_hook() or some such that is run at the start of the ctan target that could do whatever you want?

**Comment by zauguin** [comment]
I don't think that l3build should start trying to determine if something changes, especially since this is in general rather hard.

Also I know some projects (including `l3build` itself) where such a warning would always trigger (typically because `l3build tag` adds a new section to the ChangeLog).

**Comment by FrankMittelbach** [comment]
I agree with @zauguin that this should not be done automatically, but @davidcarlisle 's suggestion to provide a hook for those who want to set up something for their own projects may be a good compromise perhaps?

**Comment by Skillmon** [comment]
Yes, that indeed sounds like a good idea (and I haven't thought about side effects when `l3build tag` does more than changing version/date information).
